src/cpu/base.cc

```diff
@@ -727,8 +727,12 @@ void
 BaseCPU::scheduleSimpointsInstStop(std::vector<Counter> inst_starts)
 {
     std::string cause = "simpoint starting point found";
+    Counter last = -1;
     for (size_t i = 0; i < inst_starts.size(); ++i) {
-        scheduleInstStop(0, inst_starts[i], cause);
+        if (last != inst_starts[i]) {
+            scheduleInstStop(0, inst_starts[i], cause);
+            last = inst_starts[i];
+        }
```

Comment:
I suggest a different approach:
- Check for duplicates with `const bool hasDuplicates = std::adjacent_find(inst_starts.begin(), inst_starts.end()) != inst_starts.end();
`
- Raise a fatal if there are duplicates
- Remove duplicates in python before calling this function.

Reply:
The duplicates should be allowed because the warmup length can be larger than multiple SimPoint starting instructions. When it happens, the ```inst_starts``` for them will all be 0 and cause the duplicates.

Reply:
I don't think that's what the code is doing... if you allow duplicates then why are you checking `last != inst_starts[i]`?

Reply:
The code prevents scheduling exit events for duplicate instructions count, but it allows the list to have duplicates.

Reply:
Why do you think it's better to not remove the duplicates in python? 

I have to say, I don't like increasing the complexity of this code. I think that it would be better to not allow duplicates, but I'm open to having my mind changed.

Reply:
For example, when the SimPoints = [1, 2, 3] with SimPoint interval = 1000 and warmup length = 2000, then the `inst_starts` = [0, 0, 1000]. I want the code only schedule one exit event for instruction 0 and let the  `simpoints_save_checkpoint_generator` to deal with saving checkpoints for SimPoints with the same `inst_start`.

Reply:
I can remove the duplicates in the [base_cpu_core.py](https://gem5.googlesource.com/public/gem5/+/refs/heads/develop/src/python/gem5/components/processors/base_cpu_core.py#155) instead of here, so it doesn't increase the complexity in the C++ code and allow the list in the stdlib SimPoint module to have duplicate starting instructions.